### PR TITLE
Verify downloaded file checksum

### PR DESCRIPTION
**Pull request by TinaMor**
image

### CODE REVIEW

File: containers-toolkit/Public/ContainerdTools.psm1 [hunk 1]
```diff
@@ -81,16 +81,31 @@ function Install-Containerd {
 
             # TODO: Use downloaded file if it exists
```
Comment (billywr) [inline containers-toolkit/Public/ContainerdTools.psm1:82]: do you still need this TODO

**Review by billywr — DISMISSED** [review]

**Review by CharityKathure — APPROVED** [review]
LGTM

File: containers-toolkit/Public/BuildkitTools.psm1 [hunk 2]
```diff
@@ -15,6 +15,71 @@ function Get-BuildkitLatestVersion {
     return $latestVersion
 }
 
+function Test-BuildkitChecksum {
```
[the remark this answers is not in the document]
Comment (TinaMor) [reply containers-toolkit/Public/BuildkitTools.psm1:18]: We could use schemas when the sha is stored in a JSON file.

[the remark this answers is not in the document]
Comment (TinaMor) [reply containers-toolkit/Public/BuildkitTools.psm1:19]: @profnandaa There is no standard format for SBOM files. If a new tool is added and Checksum digest is stored in a JSON file, the dev has to add a script block to extract the Hash function and the SHA value.

**Review by profnandaa — COMMENTED** [review]
Could you consider these "generic" functions:
```
- DownloadFile($Url)
- GetChecksum($ToolName) -> depending on the tool, the way to obtain the checksum will be different => the final return value should just be a @(algo, file) string array
- VerifyFile($File, $Algo, $Checksum)
```
Then those general methods can be re-used in the tool specific installation steps, at the beginning before you proceed with the other steps.

File: containers-toolkit/Private/CommonToolUtilities.psm1 [hunk 3]
```diff
@@ -111,6 +123,294 @@ function Get-InstallationFile {
...
+
+function GenericExtractDigest {
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Downloaded tool file path")]
+        [String]$DownloadedFile,
+
+        [parameter(Mandatory = $true, HelpMessage = "JSON schema file path")]
+        [String]$ChecksumFile,
+
+        [parameter(Mandatory = $true, HelpMessage = "JSON schema file path")]
```
Comment (profnandaa) [inline containers-toolkit/Private/CommonToolUtilities.psm1:381]: nit: do we still need the schema file? can't see it in use in the code.

File: containers-toolkit/Private/CommonToolUtilities.psm1 [hunk 4]
```diff
@@ -111,6 +123,294 @@ function Get-InstallationFile {
...
+        Throw "Invalid JSON format in checksum file. $_"
+    }
+}
+
+function GenericExtractDigest {
+    param(
+        [parameter(Mandatory = $true, HelpMessage = "Downloaded tool file path")]
+        [String]$DownloadedFile,
+
+        [parameter(Mandatory = $true, HelpMessage = "JSON schema file path")]
```
Comment (profnandaa) [inline containers-toolkit/Private/CommonToolUtilities.psm1:378]: can update this `HelpMessage` for `$ChecksumFile`.

**Review by profnandaa — APPROVED** [review]
Except for the few comments.

**Review by billywr — APPROVED** [review]

**Review by profnandaa — APPROVED** [review]

**Pull request merged**